static/src/javascripts/projects/common/modules/support/supportMessaging.ts

```diff
@@ -0,0 +1,110 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type {
+	PageTracking,
+	Tag,
+} from '@guardian/support-dotcom-components/dist/shared/src/types/targeting';
+import type React from 'react';
+import { ARTICLES_VIEWED_OPT_OUT_COOKIE } from 'common/modules/commercial/user-features';
+import config from 'lib/config';
+import { getCookie } from 'lib/cookies';
+
+// See https://github.com/guardian/support-dotcom-components/blob/main/module-versions.md
+export const ModulesVersion = 'v3';
+
+export const isHosted = config.get('page.isHosted');
+
+export const supportDotcomComponentsUrl = config.get('page.isDev')
+	? `https://contributions.code.dev-guardianapis.com`
+	: `https://contributions.guardianapis.com`;
+
+export const dynamicImport = async (
+	url: string,
+	name: string,
+): Promise<React.FC<any>> => {
+	/* eslint-disable
+			@typescript-eslint/no-unsafe-assignment,
+			@typescript-eslint/no-unsafe-call,
+			@typescript-eslint/no-unsafe-member-access,
+			--
+			dynamic import
+		 */
+	// @ts-expect-error -- see dynamic-import-init.js
+	const component = await window.guardianPolyfilledImport(url);
```

Comment:
Would it be possible to add types around `window.guardianPolyfilledImport`?

Maybe for a subsequent PR.

Reply:
yes that would be good - dynamic-import-init.js would have to be migrated to TS so I'll leave it for now

Reply:
Maybe a type declaration is sufficient. See #24596